### ISSUE TITLE
Gateway tests leave gateway processes running

### ISSUE THREAD

**Issue by anjohnson** [epics-extensions/ca-gateway#10]
I ran the new gateway self-tests some time back, apparently on May 11th. My workstation has not been rebooted since then, and I just noticed a whole load of gateway processes still running that were started on that date. If the test programs can't manage to stop their own gateway processes at the end of testing, they should at least inform the user so they can be stopped manually.

**Comment by ralphlange** [comment]
Oops. 
That's on Linux?

**Comment by anjohnson** [comment]
Yup. Thanks!

**Comment by ralphlange** [comment]
I was not able to make the tests leave gateway processes behind.

If I remember correctly, you were having some trouble running the tests on that day - could it be that the gateway processes were left because of crashes of their python wrappers?
(If that's the case, even adding atexit() handlers wouldn't help.)

**Comment by anjohnson** [comment]
It happened again for me just now, because the version of pyepics we have installed is using the wrong CA library for my host architecture. My Base has both 32 and 64-bit libraries built, and pyepics was loading the first copy of libca.so that it came across, which promptly failed leaving the gateway process running. You might be able to catch this kind of error, but I don't know whether it's worth spending time on it:

```
Copying test files test.db access.txt
Using pvlist.txt for BRE
nosetests
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/pyepics-3.2.4-py2.6.egg/epics/ca.py", line 208, in initialize_libca
    libca = load_dll(dllname)
  File "/usr/lib64/python2.6/ctypes/__init__.py", line 431, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib64/python2.6/ctypes/__init__.py", line 353, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /usr/local/epics/extensions-base/3.14.12.3-ext1/lib/linux-x86/libca.so: wrong ELF class: ELFCLASS32
ETraceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/pyepics-3.2.4-py2.6.egg/epics/ca.py", line 208, in initialize_libca
    libca = load_dll(dllname)
  File "/usr/lib64/python2.6/ctypes/__init__.py", line 431, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib64/python2.6/ctypes/__init__.py", line 353, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /usr/local/epics/extensions-base/3.14.12.3-ext1/lib/linux-x86/libca.so: wrong ELF class: ELFCLASS32
...
```
Feel free to close this issue if you decide wontfix.

**Comment by ralphlange** [comment]
@anjohnson: were you able to test the fix I tried in the [PR #12](https://github.com/epics-extensions/ca-gateway/issues/12) mentioned above?

**Comment by anjohnson** [comment]
Fixed by #12
